### PR TITLE
Transform gregor OOBM for kbfs.favorites into notification

### PR DESCRIPTION
**Pull request by patrickxb**
To allow kbfs to invalidate its favorite list cache.

r? @maxtaco 

cc @strib 

Also, cc @mmaxim who is working on the API server side of this

### CODE REVIEW

**Comment by patrickxb** [comment]
@keybase/react-hackers can you fix the flow error?  added favorites to NotificationChannels in protocol.

**Comment by chrisnojima** [comment]
looking now

File: go/service/gregor.go [hunk 1]
```diff
@@ -86,14 +90,60 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
 
 func (g *gregorHandler) BroadcastMessage(ctx context.Context, m gregor1.Message) error {
 	g.G().Log.Debug("gregor handler: broadcast: %+v", m)
-	return nil
+
+	obm := m.ToOutOfBandMessage()
+	if obm == nil {
```
Comment (maxtaco) [inline go/service/gregor.go]: This isn't necessarily an error.  If it's an in-band message, than the out-of-band message field will be nil.

File: go/service/gregor.go [hunk 2]
```diff
@@ -86,14 +90,60 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
...
 func (g *gregorHandler) Shutdown() {
 	g.conn.Shutdown()
 	g.conn = nil
 }
 
+func (g *gregorHandler) kbfsFavorites(ctx context.Context, m gregor.OutOfBandMessage) error {
+	if m.Body() == nil {
+		return errors.New("gregor handler for kbfs.favorites: nil message body")
+	}
+	body, err := jsonw.Unmarshal(m.Body().Bytes())
```
Comment (maxtaco) [inline go/service/gregor.go]: Do you want to use the "proper" unmarshalling scheme that Go likes?

**Comment by maxtaco** [comment]
Had a small question but looking good.

File: go/service/gregor.go (hunk 2 above)
Comment (patrickxb) [inline go/service/gregor.go]: I would, but didn't know what the object would be yet, so just used jsonw for now.

File: go/service/gregor.go (hunk 1 above)
Comment (patrickxb) [inline go/service/gregor.go]: I assume this function will get added to once we handle other messages.

**Comment by maxtaco** [comment]
OK 👍

**Pull request merged**